src/tty/windows.rs

```diff
@@ -62,111 +50,411 @@ pub fn process_should_exit() -> bool {
+        }
+
+        (winpty, conout_pipe, conin_pipe)
+    }
+
+    impl<'a> OnResize for Winpty<'a> {
+        fn on_resize(&mut self, sizeinfo: &SizeInfo) {
+            if sizeinfo.cols().0 > 0 && sizeinfo.lines().0 > 0 {
+                self.set_size(sizeinfo.cols().0, sizeinfo.lines().0)
+                    .unwrap_or_else(|_| info!("Unable to set winpty size, did it die?"));
+            }
+        }
+    }
+}
+
+mod conpty {
+    use super::*;
+
+    use std::env;
+    use std::mem;
+    use std::ptr;
+    use std::os::windows::io::IntoRawHandle;
+
+    use dunce::canonicalize;
+    use miow;
+    use mio_anonymous_pipes::{AsyncAnonRead, AsyncAnonWrite};
+    use winapi::shared::basetsd::{PSIZE_T, SIZE_T};
+    use winapi::shared::ntdef::{HANDLE, HRESULT, LPCWSTR, LPWSTR};
+    use winapi::shared::minwindef::{BYTE, DWORD};
+    use winapi::shared::winerror::S_OK;
+    use winapi::um::libloaderapi::{GetModuleHandleA, GetProcAddress};
+    use winapi::um::processthreadsapi::*;
+    use winapi::um::winbase::{STARTUPINFOEXW, EXTENDED_STARTUPINFO_PRESENT};
+    use winapi::um::wincon::{COORD, HPCON};
+    use widestring::U16CString;
+
+    use config::{Config, Shell};
+    use display::OnResize;
+    use cli::Options;
+    use term::SizeInfo;
+
+    #[allow(non_snake_case)]
+    struct ConptyApi {
+        CreatePseudoConsole: unsafe extern "system" fn (
+            COORD,
+            HANDLE,
+            HANDLE,
+            DWORD,
+            *mut HPCON
+        ) -> HRESULT,
+        ResizePseudoConsole: unsafe extern "system" fn (HPCON, COORD) -> HRESULT
+    }
+
+    lazy_static! {
+        static ref CONPTY_API: Option<ConptyApi> = unsafe {
+            let hmodule = GetModuleHandleA("kernel32\0".as_ptr() as _);
+            assert!(!hmodule.is_null());
+
+            let cpc = GetProcAddress(hmodule, "CreatePseudoConsole\0".as_ptr() as _);
+            let rpc = GetProcAddress(hmodule, "ResizePseudoConsole\0".as_ptr() as _);
+
+            if cpc.is_null() || rpc.is_null() {
+                None
+            } else {
+                Some(ConptyApi {
+                    CreatePseudoConsole: mem::transmute(cpc),
+                    ResizePseudoConsole: mem::transmute(rpc)
+                })
+            }
+        };
+    }
+
+    // TODO: implement drop to close the pseudoconsole and child process.
+    pub struct Conpty {
+        pub handle: HPCON
+    }
+
+    unsafe impl Send for Conpty {}
+
+    pub type Agent = Conpty;
+
+    pub fn new(
+        config: &Config,
+        options: &Options,
+        size: &SizeInfo,
+        _window_id: Option<usize>,
+    ) -> Option<(Agent, AsyncAnonRead, AsyncAnonWrite)> {
+
+        if !config.enable_experimental_conpty_backend() {
+            return None
+        }
+
+        CONPTY_API.as_ref().map(|api| {
+
+            let mut pty_handle = 0 as HPCON;
+
+            // TODO: buffer size??
+            let (conout, conout_pty_handle) = miow::pipe::anonymous(10000).unwrap();
```

Comment:
This seems like quite a large default to me.

Reply:
No particular reason. I'm going to change it to be "0", which according to the docs for CreatePipe is going to let the system size it as the default buffer size. This is likely good enough, but I'll leave a comment stating we might gain something from tuning. (Though hopefully when console overlapped i/o comes along we switch to named pipes and can tune then.)